packages/next/src/server/base-server.ts

```diff
@@ -2723,19 +2717,32 @@ export default abstract class Server<
       const isProduction = !this.renderOpts.dev
       const didRespond = hasResolved || res.sent
 
+      // If we haven't found the static paths for the route, then do it now.
       if (!staticPaths) {
-        ;({ staticPaths, fallbackMode } = hasStaticPaths
-          ? await this.getStaticPaths({
-              pathname,
-              requestHeaders: req.headers,
-              isAppPath,
-              page: components.page,
-            })
-          : { staticPaths: undefined, fallbackMode: false })
+        if (hasStaticPaths) {
```

Comment:
I found it confusing to be checking that `staticPaths` is false but then checking `hasStaticPaths` is true right after, until I realized the `hasStaticPaths` variable corresponds with whether or not the component has the static paths function. (Maybe it should be renamed to `hasGetStaticPaths` or something?)